shared/dev/extractGraphQlOperationCodegenPlugin.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+
+const { visit } = require('graphql');
+const path = require('path');
+
+const ROOT_FOLDER = path.join(__dirname, '../../')
+/**
+ *
+ * @param {import('graphql').GraphQLSchema} schema
+ * @param {import('@graphql-codegen/plugin-helpers').Types.DocumentFile[]} documents
+ * @param {{interfaceNameForOperations?: string}} config
+ */
+const plugin = (schema, documents, config) => {
+  const { interfaceNameForOperations = 'AllOperations' } = config;
+
+  /** @type {{name: string, location?: string}[]} */
+  const allOperations = [];
+
+  for (const item of documents) {
+    if (item.document) {
+      visit(item.document, {
+        enter: {
+          OperationDefinition: node => {
+            if (node.name && node.name.value) {
+              allOperations.push({
+                name: node.name.value,
+                location: item.location && path.relative(ROOT_FOLDER, item.location),
+              });
+            }
+          },
+        },
+      });
+    }
+  }
+
+  const interfaceFields = allOperations.map(
+    ({ name, location }) => `
+/** ${location || 'location not found'}*/
+${name}: (variables: ${name}Variables) => ${name}Result;
+`
+  );
+  if (interfaceFields.length === 0) {
+    console.warn('No operations found');
+    // TODO maybe throw an error?
```

Comment:
Let's maybe log a warning with the operation name at least?

```suggestion
    console.warn(`No interface fields found for GraphQL operation ${JSON.stringify(name)}`)
```

Reply:
at this point we don't have a context of an operation because it is an aggregate result across all of them, but I will add the desired interface name in here